src/inference/src/system_conf.cpp

```diff
@@ -357,6 +371,16 @@ void set_cpu_used(const std::vector<int>& cpu_ids, const int used) {
     }
 }
 
+int get_socket_by_numa_node(int numa_node_id) {
+    CPU& cpu = cpu_info();
+    for (int i = 0; i < cpu._processors; i++) {
+        if (cpu._cpu_mapping_table[i][CPU_MAP_NUMA_NODE_ID] == numa_node_id) {
+            return cpu._cpu_mapping_table[i][CPU_MAP_SOCKET_ID];
+        }
+    }
+    return -1;
```

Comment:
why do we two implementations of the same method?

Reply:
I have modified the implementation on MAC OS.